### DISCUSSION TITLE
How to pull from a branch into my branch

### DISCUSSION THREAD

**Discussion by halldorr** [community/community#32354] [New to GitHub] [answered]
I am still fairly new to Github outside of the extreme basics and have run into a little problem. We have three branches we're working with: master, dev, and dev-halldorr. I created my repo from the master so my Github for Desktop always pulls from master when I pull to update my files. What I need to do is pull from dev to dev-halldorr where I will then work on my files and then I need to make a pull request to dev to get the code back there. So I'm looking for some help on how to do this. If I can do it all in Github for Desktop that would be ideal but if I need to use command line I can try to figure that out. So I am looking to do this:

1) Pull the updated files from the dev branch into my own branch called dev-halldorr
2) Make a pull request when I'm done to send my changes back to the dev branch.

**Comment by librity** [comment, marked as answer]
To get the commits from the other branch:
```bash
git checkout dev-halldorr
git merge BRANCH_YOU_WANT_TO_MERGE_WITH
```
Then you'll push those commits to you repo, and open a Pull Request from the github interface.

**Comment by Hshgrhjrhrgk** [comment]
715471806 الي يبا يدفع فلوس نشغلها يراسل واتساب خاص
